src/main/java/degree/CatalogChecker.java

```diff
@@ -0,0 +1,290 @@
+package degree;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.microsoft.azure.functions.ExecutionContext;
+
+import FuncTest.TestFunc.globals;
+import homework.LoginCredentials;
+
+public class CatalogChecker {
+	
+	ExecutionContext c;
+	List<Course> myCourses;
+	List<Course> coreCourses;
+	
+	public CatalogChecker(ExecutionContext c, LoginCredentials creds) {
+		this.c = c;
+		myCourses = (new CourseListGetter(creds, c)).getCourseList();
+		
+		if(myCourses.size() > 0) {
+			c.getLogger().info("================== GOT YOUR COURSES ====================");
+		} else {
+			c.getLogger().info("================== GOT YOUR COURSES ====================");
+			throw new RuntimeException("Failed to read course from UG\n");
+		}
+	}
+	
+	static public double sumPoints(final List<Course> list){
+		return list.stream().map(c -> Double.valueOf(c.getPoints()))
+				.reduce(Double.valueOf(0.0), (x,y) -> Double.valueOf(x.doubleValue()+y.doubleValue())).doubleValue();
+	}
+	
+	public List<Course> getMissingMandatory() throws SQLException{
+		List<Course> mandatory = getMandatory(), diff = difference(mandatory, myCourses);
+		if(myCourses.contains(new Course(234145, -1)))
```

Comment:
why do you need both branches?

Reply:
what branches, in github?

Reply:
I met both ifs

Reply:
Oh in the catalog we are working on courses 234145 and 044145 are interchangeable - however they are both parsed in as required courses, so it is to make sure that if one of them exists in your courses remove the other.